enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/RegisterAllocation.scala

```diff
@@ -218,6 +218,10 @@ object RegisterAllocation {
         incomingPipeline.newLong(name, nullable = false, CTNode)
         incomingPipeline
 
+      case MergeCreateNode(_, IdName(name), _, _) =>
+        // The variable name should already have been allocated by the NodeLeafPlan
```

Comment:
Maybe we can assert that this is the case?

Reply:
Good point, but though that should not hold back this PR any longer.